optimum/bettertransformer/transformation.py

```diff
@@ -68,12 +69,15 @@ def replace_to_bettertransformer(model, config):
                 " please pass a model that is not loaded in 8-bit.",
             )
 
-        class_name = module.__class__.__name__
-        is_bt_compatible = class_name in BETTER_TRANFORMER_LAYERS_MAPPING_DICT
-
-        if is_bt_compatible:
-            fast_module = BETTER_TRANFORMER_LAYERS_MAPPING_DICT[class_name](module, config)
-            model._modules[name] = fast_module
+        # replace the module if it is a transformer layer compatible with bettertransformer
+        if (
+            module.__class__.__name__
+            == BetterTransformerManager.BETTER_TRANSFORMER_MODEL_MAPPING[config.model_type][0]
```

Comment:
nit: I think this is not super readable, and would put that into an intermediate variable:

```
should_replace_module = module.__class__.__name__ == BetterTransformerManager.BETTER_TRANSFORMER_MODEL_MAPPING[config.model_type][0]
```

Reply:
yes